test_util/test_azure.py

```diff
@@ -0,0 +1,139 @@
+    options.location = os.getenv('AZURE_LOCATION', 'East US')
+    options.linux_user = os.getenv('AZURE_LINUX_USER', 'dcos')
+    # Prefixes must not begin with a number
+    options.agent_prefix = os.getenv('AZURE_AGENT_PREFIX', 'test' + random_id(10).lower())
+    options.master_prefix = os.getenv('AZURE_MASTER_PREFIX', 'test' + random_id(10).lower())
+    options.vm_size = os.getenv('AZURE_VM_SIZE', 'Standard_D2')
+    options.num_agents = int(os.getenv('AGENTS', '2'))
+    options.name_suffix = os.getenv('AZURE_DCOS_SUFFIX', '12345')
+    options.oauth_enabled = os.getenv('AZURE_OAUTH_ENABLED', 'false') == 'true'
+    options.vm_diagnostics_enabled = os.getenv('AZURE_VM_DIAGNOSTICS_ENABLED', 'true') == 'true'
+    options.azure_cleanup = os.getenv('AZURE_CLEANUP', 'true') == 'true'
+    options.ci_flags = os.getenv('CI_FLAGS', '')
+
+    add_env = []
+    prefix = 'TEST_ADD_ENV_'
+    for k, v in os.environ.items():
+        if k.startswith(prefix):
+            add_env.append(k.replace(prefix, '') + '=' + v)
+    options.test_cmd = os.getenv(
+        'DCOS_PYTEST_CMD', ' '.join(add_env) + " py.test -vv -s -rs -m 'not ccm'" + options.ci_flags)
+    return options
+
+
+def main():
+    options = check_environment()
+    aw = AzureWrapper(
+        options.subscription_id,
+        options.client_id,
+        options.client_secret,
+        options.tenant_id)
+    dcos_resource_group = DcosResourceGroup.deploy_acs_template(
+        azure_wrapper=aw,
+        template_uri=options.template_url,
+        location=options.location,
+        group_name=options.name,
+        public_key=options.public_ssh_key,
+        master_prefix=options.master_prefix,
+        agent_prefix=options.agent_prefix,
+        admin_name=options.linux_user,
+        oauth_enabled=options.oauth_enabled,
+        vm_size=options.vm_size,
+        agent_count=options.num_agents,
+        name_suffix=options.name_suffix,
+        vm_diagnostics_enabled=options.vm_diagnostics_enabled)
+    result = 1
+    with ExitStack() as stack:
+        if options.azure_cleanup:
+            stack.push(dcos_resource_group)
+        dcos_resource_group.wait_for_deployment()
+        t = stack.enter_context(
+            tunnel(options.linux_user, load_string(options.ssh_key_path),
+                   dcos_resource_group.outputs['masterFQDN'], port=2200))
+        result = integration_test(
+            tunnel=t,
+            dcos_dns=dcos_resource_group.get_master_ips()[0],
+            master_list=dcos_resource_group.get_master_ips(),
+            agent_list=dcos_resource_group.get_private_ips(),
+            public_agent_list=dcos_resource_group.get_public_ips(),
+            test_cmd=options.test_cmd)
+    if result == 0:
+        log.info('Test successsful!')
+    else:
+        logging.warning('Test exited with an error')
+    if options.ci_flags:
+        result = 0  # Wipe the return code so that tests can be muted in CI
```

Comment:
For tests failing and showing green that is tracked in: https://mesosphere.atlassian.net/browse/DCOS-12772

In this case I just fat fingered it so it should be working on this next push